modules/lily-compiler/src/test/java/io/github/tomboyo/lily/compiler/feature/ComposedSchemaTest.java

```diff
@@ -0,0 +1,108 @@
+package io.github.tomboyo.lily.compiler.feature;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.github.tomboyo.lily.compiler.LilyExtension;
+import io.github.tomboyo.lily.compiler.LilyExtension.LilyTestSupport;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+public class ComposedSchemaTest {
+  @Nested
+  @ExtendWith(LilyExtension.class)
+  class OneOf {
+    @BeforeAll
+    static void beforeAll(LilyTestSupport support) {
+      support.compileOas(
+          """
+            openapi: 3.0.2
+            components:
+                schemas:
+                  MySchema:
+                    oneOf:
+                      - $ref: '#/components/schemas/Foo'
+                      - type: object
+                        properties:
+                          foo:
+                            type: integer
+                            format: int32
+                      - type: string
+                      - type: string
+                        format: email
+                      - type: integer
+                        format: int32
+                  #MySchemaStringEmailAlias:
+                  #  type: string
+                  #  format: email
+                  Foo:
+                    type: string
+            """);
+    }
+
+    @Test
+    void fooImplementsInterface(LilyTestSupport support) {
+      assertTrue(
+          support.evaluate(
+              """
+                    return {{package}}.MySchema.class.isAssignableFrom(
+                        {{package}}.Foo.class);
+                    """,
+              Boolean.class));
+    }
+
+    /*
+     * The second element in the OneOf is anonymous, so we combine the root component's name with its position to create
+     * the name MySchema2. This name is unstable between schema versions: if the position of the type in the list
+     * changes, then so does the generated type name. This is unavoidable.
```

Comment:
Well, maybe this is avoidable. The type name could be derived from some kind of hash-like function. I.e. perform a DFS on the schema, keeping the first letter of each keyword, and concatenate the result (assuming that actually makes unique names; just a though).

Reply:
opened #91